File: src/896_monotonic_array/monotonic_array.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
   public:
    bool isMonotonic(vector<int>& nums) {
        bool inc = true, dec = true;
        for (int i = 1; i < nums.size(); ++i) {
            if (nums[i] > nums[i - 1]) dec = false;
            if (nums[i] < nums[i - 1]) inc = false;
        }
        return inc || dec;
    }
};

void test() {
    Solution sol;
    vector<vector<int>> tests = {{1, 2, 2, 3}, {6, 5, 4, 4}, {1, 3, 2}, {1, 1, 1}, {1}, {}};
    vector<string> expected = {"True", "True", "False", "True", "True", "True"};
    for (int i = 0; i < tests.size(); ++i) {
        cout << boolalpha << sol.isMonotonic(tests[i]) << " | " << expected[i] << endl;
    }
}

int main() {
    test();
    return 0;
}

File: src/896_monotonic_array/monotonic_array.py
from typing import List


class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        inc = dec = True
        for i in range(1, len(nums)):
            if nums[i] > nums[i - 1]:
                dec = False
            if nums[i] < nums[i - 1]:
                inc = False
        return inc or dec


def test():
    sol = Solution()
    print(sol.isMonotonic([1, 2, 2, 3]), "| True")
    print(sol.isMonotonic([6, 5, 4, 4]), "| True")
    print(sol.isMonotonic([1, 3, 2]), "| False")
    print(sol.isMonotonic([1, 1, 1]), "| True")
    print(sol.isMonotonic([1]), "| True")
    print(sol.isMonotonic([]), "| True")


if __name__ == "__main__":
    test()
